### PR TITLE
updating salmon tool

### PR DESCRIPTION
**Pull request by ramezrawas**
@bgruening @nsoranzo 
I have updated the xml file, though testing is not working yet
kindly have a look for your feedback

### CODE REVIEW

File: tools/salmon/salmon.xml [hunk 1]
```diff
@@ -143,133 +172,199 @@
...
+        <data name="output_gene_quant" format="tabular" from_work_dir="results/quant.genes.sf" label="${tool.name} on ${on_string} (Gene Quantification)">
             <filter>geneMap is True</filter>
         </data>
     </outputs>
     <tests>
         <test>
+            <param name="TranscriptSource" value="history" />
+            <param name="ownFile" value="test-data/transcripts.fasta" ftype="fasta" />
+            <param name="kmerSize" value="31" />
+            <param name="type" value="quasi" />
```
Comment (bgruening) [inline tools/salmon/salmon.xml]: You are missing an output here.

File: tools/salmon/salmon.xml [hunk 2]
```diff
@@ -144,132 +178,204 @@
...
 
     <outputs>
-        <data name="output_quant" format="tabular" from_work_dir="quant.sf" label="${tool.name} on ${on_string} (Quantification)" />
-        <data name="output_bias_corrected_quant" format="tabular" from_work_dir="quant_bias_corrected.sf" label="${tool.name} on ${on_string} (Bias corrected Quantification)">
+        <data name="output_quant" format="tabular" from_work_dir="./output/quant" label="${tool.name} on ${on_string} (Quantification)" />
+        <!-- <data name="output_bias_corrected_quant" format="tabular" from_work_dir="results/quant_bias_corrected.sf" label="${tool.name} on ${on_string} (Bias corrected Quantification)">
             <filter>biasCorrect is True</filter>
         </data>
-        <data name="output_gene_quant" format="tabular" from_work_dir="quant.genes.sf" label="${tool.name} on ${on_string} (Gene Quantification)">
+        <data name="output_gene_quant" format="tabular" from_work_dir="results/quant.genes.sf" label="${tool.name} on ${on_string} (Gene Quantification)">
```
Comment (bgruening) [inline tools/salmon/salmon.xml]: should this be ./output/quant.genes.sf or ./output/results/quant.genes.sf?

**Comment by ramezrawas** [comment]
@nsoranzo @bgruening for your kind review and feedback.

Thanx in advance

**Comment by bgruening** [comment]
Thanks @ramezrawas!

**Comment by bgruening** [comment]
ping @pavanvidem

**Comment by pavanvidem** [comment]
If all tests are working, then ready to go. Thanks!

**Pull request merged**